src/OpenRiaServices.DomainServices.Client/Test/Client.Test/Data/ServerSideAsyncTests.cs

```diff
@@ -330,5 +330,139 @@ public void Invoke_TaskAsyncWithException_InTask()
             });
             this.EnqueueTestComplete();
         }
+
+        [TestMethod]
+        [Description("Tests that an async Task Insert operation is executed and waited for")]
+        public async Task Insert_TaskAsync()
+        {
+            var ctx = new ServerSideAsyncDomainContext(TestURIs.ServerSideAsync);
+
+            var rangeItem = new RangeItem() { Text = "insert" };
+            ctx.RangeItems.Add(rangeItem);
+            await ctx.SubmitChangesAsync();
+
+            Assert.AreEqual(42, rangeItem.Id);
+        }
+
+        [TestMethod]
+        [Description("Test that exceptions thrown in Task from Task async Insert methods are propagated")]
+        public async Task Insert_TaskAsyncWithException_InTask()
+        {
+            var ctx = new ServerSideAsyncDomainContext(TestURIs.ServerSideAsync);
+
+            var rangeItem = new RangeItem2() { Text = "insert failed" };
+            ctx.RangeItem2s.Add(rangeItem);
+
+            await AssertDomainExceptionIsThrown(ctx.SubmitChangesAsync, 25);
+        }
+
+        [TestMethod]
+        [Description("Tests that an async Task Update operation is executed and waited for")]
+        public async Task Update_TaskAsync()
+        {
+            var ctx = new ServerSideAsyncDomainContext(TestURIs.ServerSideAsync);
+
+            var rangeItem = new RangeItem() { Text = "test" };
+            ctx.RangeItems.Attach(rangeItem);
+
+            rangeItem.Text = "updated";
+
+            await ctx.SubmitChangesAsync();
+        }
+
+        [TestMethod]
+        [Description("Test that exceptions thrown in Task from Task async Update methods are propagated")]
+        public async Task Update_TaskAsyncWithException_InTask()
+        {
+            var ctx = new ServerSideAsyncDomainContext(TestURIs.ServerSideAsync);
+
+            var rangeItem = new RangeItem2() { Text = "test" };
+            ctx.RangeItem2s.Attach(rangeItem);
+
+            rangeItem.Text = "updated";
+
+            await AssertDomainExceptionIsThrown(ctx.SubmitChangesAsync, 26);
+        }
+
+        [TestMethod]
+        [Description("Tests that an async Task EntityAction operation is executed")]
+        public async Task EntityAction_TaskAsync()
+        {
+            var ctx = new ServerSideAsyncDomainContext(TestURIs.ServerSideAsync);
+
+            var rangeItem = new RangeItem() { Text = "test" };
+            ctx.RangeItems.Attach(rangeItem);
+
+            rangeItem.CustomUpdateRange();
+
+            await ctx.SubmitChangesAsync();
+        }
```

Comment:
It would be good if we could verify that the entityaction was called, just as for insert